clients/android/NewsBlur/src/com/newsblur/util/ImageCache.java

```diff
@@ -52,21 +51,7 @@ public void cacheImage(String url) {
 
             File f = new File(cacheDir, fileName);
             if (f.exists()) return;
-            FileOutputStream outStream = new FileOutputStream(f);
-            int size = 0;
-            try {
-                URL u = new URL(url);
-                NetworkUtils.UrlLoadResult result = NetworkUtils.loadURL(u, outStream);
-                if (result.redirUrl != null) {
-                    // redir will only be set if the original load encountered an un-followable
-                    // redirect. Try once and only once to follow it manually.
-                    u = new URL(u, result.redirUrl);
-                    result = NetworkUtils.loadURL(u, outStream);
-                }
-                size = result.bytesRead;
-            } finally {
-                outStream.close();
-            }
+            long size = NetworkUtils.loadURL(new URL(url), f);
```

Comment:
I didn't see any missing images in my testing.

One thing I noticed was that downloading images is really slow (I don't normally have it turned on). Both the old code and the new code. Maybe it is slow storage on my N4?